modules/internal/ChapelBase.chpl

```diff
@@ -914,12 +922,15 @@ module ChapelBase {
   // statement is needed because the call to sub() will do a remote
   // fork (on) if needed.
   pragma "dont disable remote value forwarding"
+  pragma "down end count fn"
   proc _downEndCount(e: _EndCount) {
+    // inform anybody waiting that we're done
     e.i.sub(1, memory_order_release);
   }
 
   // This function is called once by the initiating task.  As above, no
   // on statement needed.
+  // called for implicit sync at end of main
```

Comment:
Pretty sure this one is called for implicit sync, explicit syncs, unbounded coforalls, and cobegins.